chore: add documentation for app icon (#45)
moxystudio/react-native-with-moxy: docusaurus/docs/welcome/instructions/app-icon.md

-   **xxxhdpi (192x192@4x):** `android/app/src/main/res/mipmap-xxxhdpi/app_icon.png`

##### Round icons:

-   **mdpi (48x48@1x):** `android/app/src/main/res/mipmap-mdpi/app_icon_round.png`
-   **hdpi (72x72@1\.5x):** `android/app/src/main/res/mipmap-hdpi/app_icon_round.png`
-   **xhdpi (96x96@2x):** `android/app/src/main/res/mipmap-xhdpi/app_icon_round.png`
-   **xxhdpi (144x144@3x):** `android/app/src/main/res/mipmap-xxhdpi/app_icon_round.png`
-   **xxxhdpi (192x192@4x):** `android/app/src/main/res/mipmap-xxxhdpi/app_icon_round.png`

After moving the assets to the following locations, there is nothing more to do.
Both "Square" and "Circle" shapes should work.

#### iOS

There are two options: either you manually move the icons and reflect the changes on the `ios/reactNativeWithMoxy/Images.xcassets/AppIcon.appiconset/Contents.json` file or
drag and drop the assets using XCode, which will automatically update it.

Using XCode:

On the project tree, navigate to `reactNativeWithMoxy > Images.xcassets`.

![XCode AppIcon](/img/xcode-app-icon.png)

Now you should be able to drag and drop each asset to its correct location.

> ![XCode AppIcon Drag and Drop](/img/xcode-app-icon-drag-n-drop.png)

Author: I should add a screenshot to demonstrate how the Xcode looks after the icons are in place

> ❗️Android 8.0 Oreo introduced the adaptive icons and apps running on it or later versions should support them.

##### Square icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Square**.

![Image Asset Studio - Launcher Icons (Legacy only) with Square shape](/img/image-asset-studio-launcher-icons-legacy-only-square.png)

After that you should click on **Next** and finally on the **Finish** button.

##### Round icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon_round`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Circle**.

![Image Asset Studio - Launcher Icons (Legacy only) with Circle shape](/img/image-asset-studio-launcher-icons-legacy-only-circle.png)

After that you should click on **Next** and finally on the **Finish** button.

> > 💡 Note that the Google Play icon won't be generated by the Image Asset Studio, so it should be manually made by following the [official design specifications](https://developer.android.com/google-play/resources/icon-design-specifications).

Author: To be honest, I don't remember about us agreeing on having a dedicated sub section. I agree with the idea anyway 👍

#### iOS

The iOS assets can be generated using [App Icon Generator](https://appicon.co/).
On the "iOS and macOS" section, only `iPhone` and `iPad` should be selected.
It is possible to either click and browse the image source or simply drag and drop it.
After selecting the image, click on the "Generate" button to download the assets.

Although not mandatory, we recommend to rename the generated assets according to the following pattern: `<resolution>@x<multiplier>`.
This way, it becomes makes easier to identify the attributes of each icon and their purpose simply by glancing at its name.

> > ℹ️ Unfortunately there isn't an official tool for app icon generation on iOS.
> > Even though there are Design related tools that solves this problem, there aren't similar ones which are non-designers friendly and significantly better than the recommended tool.

Reviewer: ```suggestion
> ℹ️  Unfortunately, there isn't an official tool to generate app icons for iOS. Even though there are design tools that solve this problem, e.g. Sketch and Adobe Illustrator, there aren't any suitable and up-to-date open source CLI tools which automatically generate all the necessary icons.
```

> > Even though there are Design related tools that solves this problem, there aren't similar ones which are non-designers friendly and significantly better than the recommended tool.

Reviewer: A quick search led me to these two developer tools:
- https://github.com/smallmuou/ios-icon-generator
- https://github.com/Fueled/ios-icon-generator

We don't need to cover them extensively, but at least reference them as an alternative.

Author: The first one I am sure it isn't suitable. The second one may be worth taking a closer look, will do that. I'm just a bit skeptical due to the last update was done almost 1 year ago

Author: The 2nd one works for our case! I will rewrite this part and mention it

---
id: app-icon
title: App icon
sidebar_label: App icon
---

Every app should have an impactul icon that stands out from all the others apps available on the user's device.

An app icon is a graphic that represents your app to users. It can:

> -   Appear in the list of apps installed on a device and on the Home screen.
> -   Represent shortcuts into your app (for example, a contact shortcut icon that opens detail information for a contact).
> -   Help users find your app on Google Play and App Store.

Reviewer: ```suggestion
- Appear in the list of apps installed on a device and on the Home screen.
- Represent shortcuts into your app (for example, a contact shortcut icon that opens detail information for a contact).
- Help users find your app on Google Play and App Store.
```

> > ℹ️ An app icon is mandatory to publish the app to TestFlight/App Store and Google Play

Reviewer: ```suggestion
> ℹ️  An app icon is mandatory to publish the app to TestFlight/App Store and Google Play.
```

### How to generate the assets

First and foremost, we need to take into consideration that automatic generation won't work everytime.
Sometimes, small adjustments to an icon may be necessary and that should probably be done manually so it looks exactly how it is supposed to.

> There are design tools that solve this problem, e.g. Sketch and Adobe Illustrator, and they can provide automatic exportation under the Designer's control and customization. Nonetheless, automatic generation should always be the first step.

Reviewer: Rephrased it a bit, but I think it looks much better already!

```suggestion
There are design tools that solve this problem, e.g. Sketch and Adobe Illustrator, which allow further customization and can assist designers in automatically exporting the icons. Nonetheless, automatic generation from a single asset should always be the first step to take as it is the most efficient and effective for most use cases.
```

Author: This looks better indeed, done 👍

> ❗️ The raw app icon asset's resolution should be 1024x1024px or bigger, so it can be used to generate both Android and iOS icons through the recommended generators.

#### Android

> > ❗️Android 8.0 Oreo introduced adaptive icons and apps running on it or later versions are ought to support them.

Reviewer: ```suggestion
> ❗️ Android 8.0 Oreo introduced adaptive icons and apps running on it or later versions are ought to support them.
```

> The Android assets can be generated using [Image Asset Studio](https://developer.android.com/studio/write/image-asset-studio).

Reviewer: ```suggestion
Android assets can be generated using [Image Asset Studio](https://developer.android.com/studio/write/image-asset-studio).
```

To start Image Asset Studio, follow these steps:

1. Open the project on Android Studio.
2. In the **Project** window, select the [Android view](https://developer.android.com/studio/projects#ProjectFiles).
3. Right-click the **res** folder and select **New > Image Asset**.

![Image Asset Studio](/img/image-asset-studio.png)

The app icon should have both a square and a round version.
The tool offers a few options which you can tweak as much as you like, but we only cover the necessary ones.

> > ℹ️ The following instructions won't cover how to generate [adaptive icons](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive), which are icons that can display a variety of shapes across different device models.

Reviewer: ```suggestion
> ℹ️  The following instructions don't cover how to generate [adaptive icons](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive), which are icons that can display a variety of shapes across different device models.
```

##### Square icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Square**.

![Image Asset Studio - Launcher Icons (Legacy only) with Square shape](/img/image-asset-studio-launcher-icons-legacy-only-square.png)

After that you should click on **Next** and finally on the **Finish** button.

##### Round icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon_round`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Circle**.

![Image Asset Studio - Launcher Icons (Legacy only) with Circle shape](/img/image-asset-studio-launcher-icons-legacy-only-circle.png)

After that you should click on **Next** and finally on the **Finish** button.

##### Google Play icon:

Having a proper Google Play icon can help your app attract new users.
This icon won't replace your app's launcher icon, it will be used in various locations on Google Play instead.

> > 💡 Note that the Google Play icon won't be generated by the Image Asset Studio, so it should be manually made by following the [official design specifications](https://developer.android.com/google-play/resources/icon-design-specifications).

Reviewer: ```suggestion
> 💡 Note that the Google Play icon isn't generated by the Image Asset Studio, so it should be manually created by following the [official design specifications](https://developer.android.com/google-play/resources/icon-design-specifications).
```

> However it isn't much different, actually. To generate those, you should use the **Launcher Icons (Adaptive and Legacy)** option for the **Icon Type**.

##### Square icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Square**.

![Image Asset Studio - Launcher Icons (Legacy only) with Square shape](/img/image-asset-studio-launcher-icons-legacy-only-square.png)

After that you should click on **Next** and finally on the **Finish** button.

##### Round icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon_round`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Circle**.

![Image Asset Studio - Launcher Icons (Legacy only) with Circle shape](/img/image-asset-studio-launcher-icons-legacy-only-circle.png)

After that you should click on **Next** and finally on the **Finish** button.

> ##### Google Play icon:

Reviewer: ```suggestion
##### Google Play icon
```

Having a proper Google Play icon can help your app attract new users.
This icon won't replace your app's launcher icon, it will be used in various locations on Google Play instead.

> 💡 Note that the Google Play icon won't be generated by the Image Asset Studio, so it should be manually made by following the [official design specifications](https://developer.android.com/google-play/resources/icon-design-specifications).

How to upload the icon:

1. Sign in to your [Play Console](https://play.google.com/apps/publish/).
2. Click **All Applications**.
3. Select an app.
4. On the left menu, click **Store Presence > Store listing**.
5. Under "Graphic Assets" you can upload your images, including the Google Play icon (Hi-res icon).

#### iOS

> The iOS assets can be generated using [App Icon Generator](https://appicon.co/).

Reviewer: ```suggestion
iOS assets can be generated using [App Icon Generator](https://appicon.co/).
```

> On the "iOS and macOS" section, only `iPhone` and `iPad` should be selected.

Reviewer: ```suggestion
In the "iOS and macOS" section, only `iPhone` and `iPad` should be selected.
```

-   **hdpi (72x72@1\.5x):** `android/app/src/main/res/mipmap-hdpi/app_icon.png`
-   **xhdpi (96x96@2x):** `android/app/src/main/res/mipmap-xhdpi/app_icon.png`
-   **xxhdpi (144x144@3x):** `android/app/src/main/res/mipmap-xxhdpi/app_icon.png`
-   **xxxhdpi (192x192@4x):** `android/app/src/main/res/mipmap-xxxhdpi/app_icon.png`

##### Round icons:

-   **mdpi (48x48@1x):** `android/app/src/main/res/mipmap-mdpi/app_icon_round.png`
-   **hdpi (72x72@1\.5x):** `android/app/src/main/res/mipmap-hdpi/app_icon_round.png`
-   **xhdpi (96x96@2x):** `android/app/src/main/res/mipmap-xhdpi/app_icon_round.png`
-   **xxhdpi (144x144@3x):** `android/app/src/main/res/mipmap-xxhdpi/app_icon_round.png`
-   **xxxhdpi (192x192@4x):** `android/app/src/main/res/mipmap-xxxhdpi/app_icon_round.png`

After moving the assets to the following locations, there is nothing more to do.

#### iOS

There are two options: either you manually move the icons and reflect the changes on the `ios/reactNativeWithMoxy/Images.xcassets/AppIcon.appiconset/Contents.json` file or
drag and drop the assets using XCode, which will automatically update it.

Using XCode:

> On the project tree, navigate to `reactNativeWithMoxy > Images.xcassets`.

Reviewer: ```suggestion
In the project tree, navigate to `reactNativeWithMoxy > Images.xcassets`.
```

#### Android

The Android assets can be generated using [Image Asset Studio](https://developer.android.com/studio/write/image-asset-studio).

> To start the Asset Studio, follow these steps:

Reviewer: ```suggestion
To start Image Asset Studio, follow these steps:
```

1. Open the project on Android Studio.
2. In the **Project** window, select the [Android view](https://developer.android.com/studio/projects#ProjectFiles).
3. Right-click the **res** folder and select **New > Image Asset**.

![Image Asset Studio](/img/image-asset-studio.png)

The app icon should have both a square and a round version.

> There tool offers a few options which you can tweak as much as you like, but we will only cover the necessary ones.

Reviewer: ```suggestion
There tool offers a few options which you can tweak as much as you like, but we only cover the necessary ones.
```

Author: I don't know why I wrote "There" this doesn't make sense! Will use your suggestion and fix this at the same time

##### Square icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Square**.

![Image Asset Studio - Launcher Icons (Legacy only) with Square shape](/img/image-asset-studio-launcher-icons-legacy-only-square.png)

After that you should click on **Next** and finally on the **Finish** button.

##### Round icons:

-   **Icon Type** should be the **Launcher Icons (Legacy only)** option.
-   **Name** should be renamed from `ic_launcher` to `app_icon_round`.
-   **Asset Type** should be **Image** and the raw app icon path should be given.
-   **Shape** should be **Circle**.

![Image Asset Studio - Launcher Icons (Legacy only) with Circle shape](/img/image-asset-studio-launcher-icons-legacy-only-circle.png)

After that you should click on **Next** and finally on the **Finish** button.

> > ℹ️ Although [Android Asset Studio](https://github.com/romannurik/AndroidAssetStudio) was also considered, we decided to adopt [Image Asset Studio](https://developer.android.com/studio/write/image-asset-studio) because it's the official tool from Google, it's built in into Android Studio and it not only generates the assets but it also places them in the correct folders.

Reviewer: I don't think there is no need to mention this.

> #### iOS

Reviewer: I think we should put the following

> Note that the filenames of the generated assets should be renamed according to the following pattern: `<resolution>@x<multiplier>`

inside a quote, add the ℹ️  emoji and probably rephrase to the following:

> ℹ️  Although not mandatory, we recommend to rename the generated assets according to the following pattern: `<resolution>@x<multiplier>`. This way, it becomes makes easier to identify the attributes of each icon and their purpose simply by glancing at its name.

The app icon should have both a square and a round version.
There tool offers a few options which you can tweak as much as you like, but we will only cover the necessary ones.

> > ℹ️  The following instructions won't cover how to generate [adaptive icons](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive), which are icons that can display a variety of shapes across different device models.

Author: I still need to add add some information about from which Android version the adaptative icons are fully supported and may be necessary, since some OEMs may need them due to the way they manipulate the icons

#### Android

> > 💡 Note that if you used [Image Asset Studio](https://developer.android.com/studio/write/image-asset-studio), you won't have to bother about this, because the tool automatically places the assets into the correct locations.

Reviewer: There is no need to link Image Asset Studio again, I guess.

```suggestion
> 💡 Note that if you use Image Asset Studio, you won't have to bother about this, because the tool automatically places the assets into the correct locations.
```

##### Square icons:

-   **mdpi (48x48@1x):** `android/app/src/main/res/mipmap-mdpi/app_icon.png`

> -   **hdpi (72x72@1\.5x):** `android/app/src/main/res/mipmap-hdpi/app_icon.png`

Reviewer: Do really need to escape `.`? 🤔

Author: Yes, we do

-   **xhdpi (96x96@2x):** `android/app/src/main/res/mipmap-xhdpi/app_icon.png`
-   **xxhdpi (144x144@3x):** `android/app/src/main/res/mipmap-xxhdpi/app_icon.png`
-   **xxxhdpi (192x192@4x):** `android/app/src/main/res/mipmap-xxxhdpi/app_icon.png`

##### Round icons:

-   **mdpi (48x48@1x):** `android/app/src/main/res/mipmap-mdpi/app_icon_round.png`

> -   **hdpi (72x72@1\.5x):** `android/app/src/main/res/mipmap-hdpi/app_icon_round.png`

Reviewer: Do really need to escape `.`? 🤔

Author: Yes, we do

-   **xhdpi (96x96@2x):** `android/app/src/main/res/mipmap-xhdpi/app_icon_round.png`
-   **xxhdpi (144x144@3x):** `android/app/src/main/res/mipmap-xxhdpi/app_icon_round.png`
-   **xxxhdpi (192x192@4x):** `android/app/src/main/res/mipmap-xxxhdpi/app_icon_round.png`

After moving the assets to the following locations, there is nothing more to do.

> Both "Square" and "Circle" shapes should work.

Reviewer: I think we can leave out this sentence. It doesn't add much.
